src/main/java/org/jabref/gui/LibraryTab.java

```diff
@@ -235,22 +235,31 @@ public void onDatabaseLoadingFailed(Exception ex) {
         dialogService.showErrorDialogAndWait(title, content, ex);
     }
 
-    public void feedData(BibDatabaseContext bibDatabaseContext) {
+    public void feedData(BibDatabaseContext bibDatabaseContextFromParserResult) {
         cleanUp();
 
-        this.bibDatabaseContext = Objects.requireNonNull(bibDatabaseContext);
+       // this.bibDatabaseContext.getDatabase().insertEntries(bibDatabaseContextFromParserResult.getEntries());
+
         // When you open an existing library, a library tab with a loading animation is added immediately.
         // At that point, the library tab is given a temporary bibDatabaseContext with no entries.
         // This line is necessary because, while there is already a binding that updates the active database when a new tab is added,
         // it doesn't handle the case when a library is loaded asynchronously.
-        stateManager.setActiveDatabase(bibDatabaseContext);
+        stateManager.setActiveDatabase(bibDatabaseContextFromParserResult);
 
-        bibDatabaseContext.getDatabase().registerListener(this);
-        bibDatabaseContext.getMetaData().registerListener(this);
+        Optional<BibDatabaseContext> x = stateManager.getOpenDatabases().stream().filter(ctx -> ctx.equals(this.bibDatabaseContext)).findFirst();
+
+        x.ifPresent(s -> stateManager.getOpenDatabases().remove(s));
+
+        this.bibDatabaseContext = Objects.requireNonNull(bibDatabaseContextFromParserResult);
+
+        stateManager.getOpenDatabases().add(bibDatabaseContextFromParserResult);
+
+        bibDatabaseContextFromParserResult.getDatabase().registerListener(this);
+        bibDatabaseContextFromParserResult.getMetaData().registerListener(this);
```

Comment:
This is duplicate of LibraryTab's constructor lines 138

Proposal:

Two LibraryTab  constructors. One with BibDatabaseContext and one without. Add a new method "setBibDatabaseContext" including the initialization of listeners etc.

Following listeners seem to be missing in the current `feedData`:

```java
        this.getDatabase().registerListener(new SearchListener());
        this.getDatabase().registerListener(new IndexUpdateListener());
        this.getDatabase().registerListener(new EntriesRemovedListener());

        // ensure that at each addition of a new entry, the entry is added to the groups interface
        this.bibDatabaseContext.getDatabase().registerListener(new GroupTreeListener());
        // ensure that all entry changes mark the panel as changed
        this.bibDatabaseContext.getDatabase().registerListener(this);

        this.getDatabase().registerListener(new UpdateTimestampListener(preferencesService));
```

Reply:
The new constructor should have `BackgroundTask` as parameter (instead of BibDatabaseContext). Maybe, an executor also has to be provided.

Reply:
I just moved the lines down... And as we replace the existing object the listeners need to be replaced as well

Reply:
We really should eventually rework the groups panel and the OpenDatabase mechanism working with CompletableFutures.

Reply:
Completable Futures aren't that different from our BackgroundTask